### PR TITLE
Add metering_report_json to GetMeteringReportResponse [DPP-1104]

### PR DESCRIPTION
**Pull request by simonmaxen-da**
For background see:
https://docs.google.com/document/d/148v-F_SmbgN8eQU4U-EjL5zvHqAjhM7qasdj5gwdcws/edit#

changelog_begin
changelog_end

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto [hunk 1]
```diff
@@ -50,6 +50,9 @@ message GetMeteringReportResponse {
 
     // The time at which the report was computed.
     google.protobuf.Timestamp report_generation_time = 3;
+
+    // The metering report json
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto:54]:
```suggestion
    // The metering report as JSON. Provided by all servers that support rendering the ``ParticipantMeteringReport`` to JSON.
```

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto [hunk 2]
```diff
@@ -50,6 +50,9 @@ message GetMeteringReportResponse {
...
+    string metering_report_json = 4;
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto:55]:
```suggestion
    google.protobuf.Struct metering_report_json = 4;
```
Use https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Struct as per https://groups.google.com/g/protobuf/c/1772tpAr3N4

Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto:55]: This allows tools like `grpcurl` to properly render the JSON data.

**Review by meiersi-da — APPROVED** [review]
Thanks. Please review the proposed changes before merging.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto (hunk 1 above)
Comment (simonmaxen-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto:55]: The protobuf ParticipantMeteringReport is not the same format as the output metering report json.  Do you think it is worth adding a Json Schema for the report format ?

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto (hunk 2 above)
Comment (simonmaxen-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto]: Yes it does.

# With string
```
{
  "request": {
    "from": "2020-01-01T00:00:00Z"
  },
  "participant_report": {
    "participant_id": "default"
  },
  "report_generation_time": "2022-07-21T17:33:38.579640Z",
  "metering_report_json": "{\"participantId\":\"default\"}"
}
```

# With struct
```
{
  "request": {
    "from": "2020-01-01T00:00:00Z"
  },
  "participant_report": {
    "participant_id": "default"
  },
  "report_generation_time": "2022-07-22T08:40:03.955245Z",
  "metering_report_json": {
      "participantId": "default"
    }
}
```

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto (hunk 1 above)
Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto:55]: I think what's missing in particular is a clarification as to why the report is returned in two formats; and which report should be used by clients of the API. Adding a description or perhaps better a link to the description of the JSON format sounds very helpful as well. Could you make an attempt at adding such commentary?

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto (hunk 2 above)
Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto]: Looks good. This can now be resolved can it?

Comment (simonmaxen-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto]: Yes, the about was output from `grpcurl`

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto (hunk 1 above)
Comment (simonmaxen-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto:55]: Agree. I've added https://digitalasset.atlassian.net/browse/DPP-1133 to cover this work.

**Pull request merged**